### PR TITLE
Docs: HighLevelRestClient#multiSearch

### LINKED ISSUE #28389: Java high-level REST client docs
Few of the API supported by the high-level REST client, do not have docs  :

- [x] ping (#29070)
- [x] exists (#29073)
- [x] multi get (#29095)
- [ ] multi search (#29144)
~~~- [ ] ranking evaluation~~~ (tracked by #28694)

Relates to #27205

### PR DESCRIPTION
**Pull request by nik9000**
Adds docs for `HighLevelRestClient#multiSearch`. Unlike the `multiGet`
docs these are much more sparse because multi-search doesn't support
setting many options on the `MultiSearchRequest` and instead just wraps
a list of `SearchRequest`s.

Closes #28389

### CODE REVIEW

**Comment by nik9000** [comment]
I marked this as closing #28389 because @cbuescher will use #28694 to track adding High Level REST Client docs to for the rank/eval API.

File: docs/java-rest/high-level/search/multi-search.asciidoc [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+The `MultiSearchRequest` is built empty and you add all of the searches that
+you wish to execute to it:
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/SearchDocumentationIT.java[multi-search-request-basic]
+--------------------------------------------------
+<1> Create an empty `MultiSearchRequest`.
+<2> Create an empty `SearchRequest` and populate it just like you
+would dfor a regular <<java-rest-high-search,`search`>>.
```
Comment (javanna) [inline docs/java-rest/high-level/search/multi-search.asciidoc]: s/dfor/for

File: docs/java-rest/high-level/search/multi-search.asciidoc [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+<3> Add the `SearchRequest` to the `MultiSearchRequest`.
+<4> Build a second `SearchRequest` and add it to the `MultiSearchRequest`.
+
+===== Optional arguments
+
+The `SearchRequests` inside of `MultiSearchRequest` support all of
```
Comment (javanna) [inline docs/java-rest/high-level/search/multi-search.asciidoc]: maybe you want the final s in SearchRequests out of the backticks ?

**Review by javanna — APPROVED** [review]
left two nits, LGTM otherwise. thanks @nik9000 !

File: docs/java-rest/high-level/search/multi-search.asciidoc (hunk 1 above)
Comment (nik9000) [reply docs/java-rest/high-level/search/multi-search.asciidoc]: Thanks!

File: docs/java-rest/high-level/search/multi-search.asciidoc (hunk 2 above)
Comment (nik9000) [reply docs/java-rest/high-level/search/multi-search.asciidoc]: Right!

**Pull request merged**